include/mqtt/mqtt_message_parser.h

```diff
@@ -0,0 +1,94 @@
+#ifndef __MQTT_MESSAGE_PARSER_H__
+#define __MQTT_MESSAGE_PARSER_H__
+
+#include "data/array.h"
+#include "utils/compile_time_checks.h"
+#include "parsing_functions.h"
+
+#include <string.h>
+#include <stdlib.h>
+
+using MessageParserParseFunction = int(*)(const char * str, char * buffer);
+
+/**
+ * \brief MQTT message parser
+ *
+ * This class provides a generic interface to parse message received over MQTT, given that these messages follow this format:
+ * [value][delimiter][value][delimiter][value]....
+ *
+ * In this message, the delimiter must be one character long, for example, a comma. The delimiter must also be consistend within
```

Comment:
The delimiter should be the same throughout the message. or consistend -> consistent.

Reply:
Both resolved